src/assignments/yunfeng/advent_of_code/2018/day_2/code.clj

```diff
@@ -0,0 +1,58 @@
+(ns advent-of-code.2018.day-2.code
+  (:require [clojure.java.io :as io]))
+
+(defn frequency-accumulator
+  "Return a new map of format like {:2 x :3 y} and help counter-for-23 to accumulate the result"
+  [frequency-for-23 new-frequency]
+  (let [frequency-vec (vals new-frequency)]
+    (merge-with + frequency-for-23 {:2 (if (some #(= 2 %) frequency-vec) 1 0) :3 (if (some #(= 3 %) frequency-vec) 1 0)})))
+
+
+(defn counter-for-23
+  "Return a map of format like {:2 x :3 y} where x means the number of ID which contains exactly 2 of any letter. Same meaning for y."
+  [reader]
+  (reduce frequency-accumulator  {:2 0 :3 0} (map frequencies (line-seq reader))))
+
+
+(defn part-1
+  [file-path]
+  (with-open [reader (io/reader file-path)]
+    (let [counter-result (counter-for-23 reader)]
+      (* (:2 counter-result) (:3 counter-result)))))
+
+(part-1 "/Users/apple/Desktop/clojure-noob/src/advent_of_code/2018/day_2/input.txt")
+
+
+(defn differ-by-one
```

Comment:
See if you can rewrite this by using functional/streams idioms:

e.g. look into map, filter, count.